lib/portage/dbapi/bintree.py

```diff
@@ -542,9 +601,14 @@ def move_ent(self, mylist, repo_match=None):
 			new_path = self.getname(mynewcpv)
 			self._pkg_paths[
 				self.dbapi._instance_key(mynewcpv)] = new_path[len(self.pkgdir)+1:]
-			if new_path != mytbz2:
-				self._ensure_dir(os.path.dirname(new_path))
-				_movefile(tbz2path, new_path, mysettings=self.settings)
+			if binpkg_path.endswith(SUPPORTED_XPAK_EXTENSIONS):
+				if new_path != mytbz2:
```

Comment:
Why are you comparing `mytbz2` here and path below?

Reply:
Because old code did, it seems check if the package name changed, move old binpkg under new name.

Reply:
The `new_path != mytbz2` comparison is always False. Since we're touching this line, we may as well fix it to use `new_path != binpkg_path` like we do for the gpkg case.